Sources/Build/BuildPlan.swift

```diff
@@ -1313,3 +1321,10 @@ struct SwiftBackDeployLibrariesNote: DiagnosticData {
             $0 <<< "Swift compiler no longer supports statically linking the Swift libraries. They're included in the OS by default starting with macOS Mojave 10.14.4 beta 3. For macOS Mojave 10.14.3 and earlier, there's an optional Swift library package that can be downloaded from \"More Downloads\" for Apple Developers at https://developer.apple.com/download/more/"
     })
 }
+
+private extension ResolvedTarget {
+    var dependsOnAnySystemModules: Bool {
+        let dependsOnAnySystemModules = !dependencies.filter { $0.target?.dependsOnAnySystemModules == true }.isEmpty
```

Comment:
This needs to be recursive, right?

Reply:
It is :)

Reply:
Oh, this won't take you across product boundaries and will be expensive. We already iterate recursive dependencies in the `plan(swiftTarget:)`/`plan(clangTarget:)` methods. You can add the flag there.